src/cookbook/gradual-composition-api-migration.md

```diff
@@ -0,0 +1,245 @@
+    <p>This is isShowing:</p>
+    <p>{{ isShowing }}</p>
+    <button @click="toggleShow">Toggle It</button>
+  </div>
+</template>
+
+<script>
+import { toggle } from '@/mixins/toggle'
+
+export default {
+  mixins: [toggle]
+}
+</script>
+```
+
+We could refactor this by creating a `features` folder and create `features/useToggle.js` instead. We suggest using a directory named `features` so that you can communicate that this is being used slightly differently from a component, it's reusable logic that you can consume.
+
+```js
+export function useToggle() {
+  const state = reactive({
+    isShowing: false
+  })
+
+  function toggleShow() {
+    state.isShowing = !state.isShowing
+  }
+
+  return {
+    ...toRefs(state),
+    toggleShow
+  }
+}
+```
+
+And we can refactor our component as follows:
+
+```vue
+<template>
+  <div id="app">
+    <h2>Togglin'</h2>
+    <p>This is isShowing:</p>
+    <p>{{ isShowing }}</p>
+    <button @click="toggleShow">Toggle It</button>
+  </div>
+</template>
+
+<script>
+import { useToggle } from '@/features/useToggle.js'
+
+export default {
+  setup() {
+    const { isShowing, toggleShow } = useToggle()
+
+    return {
+      isShowing,
+      toggleShow
+    }
+  }
+}
+</script>
+```
+
+<iframe src="https://codesandbox.io/embed/refactor-mixin-to-composition-api-hnloh?fontsize=14&hidenavigation=1&theme=dark"
+     style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
+     title="Refactor Mixin to Composition API"
+     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
+     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
+   ></iframe>
+
+Note that the template stays the same, but we've extracted the logic to use in our script section differently. If, from here, we wanted to use `isShowing` in the Options API, we could access it with `this.isShowing` now, just as we normally do with a data property, and similarly, we can access `this.toggleShow` like we would a method.
+
+You may notice we used `reactive` here instead of `refs`. This is intentional- though this example is slim, if you're refactoring a codebase, potentially full of **many** values, `reactive` translates faster and more directly as the API is closer to Options, you're still using that same object notation. We are using `toRefs` here for the purposes of destructuring.
+
+## In Place of Vuex
+
+It is possible to use the Composition API in place of Vuex, and save yourself a dependency. That said, it's not exactly necessary, and like many approaches, there are tradeoffs.
+
+If you're using Vuex, it's very clear exactly what centralized state is being used across the application. In addition, you gain the benefit of using a standardized pattern that comes with officially supported tooling in the DevTools which can make a big difference when it comes to debugging code.
```

Comment:
```suggestion
If you're using Vuex, it's very clear exactly what centralized state is being used across the application. In addition, you gain the benefit of using a standardized pattern that comes with officially supported tooling in the DevTools. This can make a big difference when it comes to debugging code.
```